### PR TITLE
[OS-1019] Migrating to new KW API: login and token functions

### PR DESCRIPTION
**Pull request by skarbat**
This PR migrates the login and token query functions to the new KW API.
I added tests to run on the RPi, which all pass.
This change provides a transparent login migration for the `kano-world-cli`.

### CODE REVIEW

File: kano_world/functions.py [hunk 1]
```diff
@@ -16,24 +16,23 @@
 from .connection import request_wrapper, content_type_json
 from .session import KanoWorldSession
 
+import mercury
 
 glob_session = None
+http_verbose = False
+kw_url = 'https://worldapi-dev.kano.me'
```
Comment (radujipa) [inline kano_world/functions.py]: Should we not use `API_URL` from `kano_world.config` instead? We can update the URL there.

File: kano_world/functions.py [hunk 2]
```diff
@@ -16,24 +16,23 @@
...
     password = password.strip()
 
-    payload = {
-        'email': login_name,
-        'password': password
-    }
-
-    success, text, data = request_wrapper('post', '/auth', data=json.dumps(payload), headers=content_type_json)
+    kw = mercury.KanoWorld(kw_url)
+    success = kw.login (login_name, password, verbose)
```
Comment (radujipa) [inline kano_world/functions.py]:
```suggestion
    success = kw.login(login_name, password, verbose)
```

File: debian/changelog [hunk 3]
```diff
@@ -1,6 +1,7 @@
 kano-profile (4.2.1-0) unstable; urgency=low
 
   * Removed dependency on kano-settings, fix dependency loop
+  * Added migration to new Kano World Services API
```
Comment (radujipa) [inline debian/changelog]: Let's bump kano-profile to `4.3.0-0`

File: kano_world/functions.py (hunk 1 above)
Comment (tombettany) [reply kano_world/functions.py]: 👍

File: kano_world/functions.py [hunk 4]
```diff
@@ -122,7 +121,11 @@ def is_registered():
 
 
 def has_token():
-    return 'token' in load_profile()
+    kw = mercury.KanoWorld(kw_url)
+    if not kw.load_data():
+        return None
+
+    return kw.get_token()
```
Comment (tombettany) [inline kano_world/functions.py]: Looks like this function used to simply return whether or not a token exists, not what it actually is so methinks that we only care about this and this function should simply return a boolean.

Also, we might want to put the `load_data()` into the constructor of the `KanoWorld()` class so we don't need to do it here.

File: kano_world/functions.py [hunk 5]
```diff
@@ -223,10 +213,7 @@ def get_mixed_username():
 
 
 def get_token():
-    try:
-        return load_profile()['token']
-    except Exception:
-        return ''
+    return has_token()
```
Comment (tombettany) [inline kano_world/functions.py]: This implementation should be what you already had for `has_token()` and the `has_token()` simply returns a boolean.

File: kano_world/functions.py (hunk 1 above)
Comment (skarbat) [reply kano_world/functions.py]: Fixed

File: kano_world/functions.py (hunk 2 above)
Comment (skarbat) [reply kano_world/functions.py]: Done

File: kano_world/functions.py (hunk 4 above)
Comment (skarbat) [reply kano_world/functions.py]: You're absolutely right - fixed and tested.

File: kano_world/functions.py (hunk 5 above)
Comment (skarbat) [reply kano_world/functions.py]: same as above, also tested

**Pull request merged**